### PR TITLE
Update cedar dependency to 4.1.0 and add new utility methods

### PR DESCRIPTION
**Pull request by davemssavage**
- Update cedar dependency to 4.1.0 this causes breaking changes in input schema so also upgrading cedarpy version to 1.0.0 as per semantic versioning guidelines
- Add new methods to convert cedar policies to/from cedar natiive and json representation

### CODE REVIEW

**Comment by skuenzli** [comment]
Thank you David! We really appreciate the PR and will take a look this week.

File: Cargo.toml [hunk 1]
```diff
@@ -11,9 +11,9 @@ crate-type = ["cdylib"]
 [dependencies]
 pyo3 = "0.19.0"
 anyhow = "1.0"
-cedar-policy = "~2.2.0"
-cedar-policy-cli = "~2.2.0"
-cedar-policy-formatter = "~2.2.0"
+cedar-policy = "~4.1.0"
```
Comment (davemssavage) [inline Cargo.toml:14]: There is a 4.2.2 release available on https://docs.rs/cedar-policy/latest/cedar_policy/ however the public docs https://docs.cedarpolicy.com/ currently refers to 4.1 hence this choice

**Comment by skuenzli** [comment]
David - thank you for putting all this together.  I hadn't anticipated a change to Cedar's schema.  So I need to learn more about that change and think about how to handle it.  Thank you for providing references.

Just an fyi... I will try to take a look at this soon and provide some concrete feedback and get the change on its way to merging.

I'm headed to re:Invent.  I'd love to connect if you'll be there.

**Comment by davemssavage** [comment]
Yeah unfortunately not going to re:Invent, no worries not urgent

File: Cargo.toml (hunk 1 above)
Comment (skuenzli) [reply Cargo.toml:14]: Thanks for thinking about the versioning.

I think we might jump to a 4.x version number to communicate support for Cedar 4.x

It looks like CedarJavaFFI did that:
https://github.com/cedar-policy/cedar-java/blob/main/CedarJavaFFI/Cargo.toml#L9

Not asking you to do anything here. Just FYI..

File: cedarpy/__init__.py [hunk 2]
```diff
@@ -158,3 +158,24 @@ def format_policies(policies: str,
...
+
+def policies_from_json_str(policies: str) -> str:
+    """Convert a json cedar policy file to a cedar policy file.
+
+    :param policies is a str containing the policies to be converted
+
+    :returns the cedar formatted policy
+    :raises ValueError: if the input policies cannot be parsed
+    """
+    return _internal.policies_from_json_str(policies)
```
Comment (skuenzli) [inline cedarpy/__init__.py:181]: Thanks for adding these functions.

Do you manage your Cedar policies in JSON?

Comment (davemssavage) [reply cedarpy/__init__.py:181]: It was to give some sort of parity with the Rust API. In Rust it's possible to parse the policy to the API objects and use this to check what has been generated, e.g. if you want to do some checking on the policy that is not supported by the schema. However returning the Cedar API objects as python objects seemed pretty complex. By allowing cedar policies to be converted to JSON it's probably good enough to be able to see what was generated and do some external validation, the other method to go from json back to cedar native just seemed sensible if you allow from native to json.

Comment (davemssavage) [reply cedarpy/__init__.py:181]: Thinking about it's probably worth adding a note to the docstring to warn others of this detail

**Comment by skuenzli** [comment]
Overall, these changes look great to me. I am going to try and mirror Cedar's conventions around branch naming.

So will merge this into a `dev/4.1.x` branch for some tweaking, then `main`, then eventually a `release/4.1.x` for maintenance.

Already branched `main` into `release/2.2.x`.

**Comment by skuenzli** [comment]
Thank you David!

**Pull request merged**